### DISCUSSION TITLE
Method for dedicated server to read all intra-company chat

### DISCUSSION THREAD

**Discussion by Azusa257** [OpenTTD/OpenTTD#10508] [Q&A]
As the title asks, is there any method for the dedicated server to read all in-game chat regardless of it being done in a private chat within a company?

**Comment by TrueBrain** [comment]
Define "method"? What are you looking for exactly?

Server-logs?
Admin-port?
Source?

I ask, as I don't want to write a reply and you go: no, I was looking for ... :D

For context, all chat is always relayed via the server, so at some point, the server does see all chats.

**Reply by Azusa257** [reply]
Server logs on the server cli preferably, since it's more visual that way. However, if I may, I would also like to know about the admin port functionality as well, mind you providing me with a reference on that?

**Comment by TrueBrain** [comment]
Okay:

Currently, the server doesn't log messages it is relaying at all. I guess an argument can be made that this is for "privacy", but I doubt that will be a strong argument to hold. That is, as long as there is no encryption done, it is a facade, to call it "privacy concerns" :)

https://github.com/OpenTTD/OpenTTD/blob/c73b88ddcaeb13025aa62a39e5e8f7e35c6872eb/src/network/network_server.cpp#L1185 processes incoming chats, and relays it to the right client. One could add a `Debug(net, 0, ...);` line there for a self-compiled Dedicated server to see what is going on.

Currently we have no other way to see what chats are being relayed. But it really depends on what you are after.

The main reason I can see that dedicated servers would like to see all chats going to and from clients/companies, is abuse / bullying. And that is why I was asking about admin-port, as I could see a place for the admin-port there.

But, people will argue privacy concerns, and I get that. But given we also have a lot of younger people playing this game .. I am not sure what is more important in the balance. Anyway, now I am just rambling, and not actually answering your question.

So in short, a dedicated server:

- Will print public chat on the console
- Will not print company / private chat in the logs nor in the console, nor is it available in the admin port
- A small change in the code will make it show up in the logs, but it does require a custom build
- I think we should consider a way to deliver this via the admin port, possibly by making it more clear to the end-user that the server-owner always sees all chat.
